aiplatform/api/AIPlatform.Samples/PredictCodeChatSample.cs

```diff
@@ -0,0 +1,106 @@
+using System.Collections.Generic;
+
+public class PredictCodeChatSample
+{
+    public string PredictCodeChat(
+        string projectId = "your-project-id",
+        string locationId = "us-central1",
+        string publisher = "google",
+        string model = "codechat-bison@001"
+    )
+    {
+        // Initialize client that will be used to send requests.
+        // This client only needs to be created once,
+        // and can be reused for multiple requests.
+        var client = new PredictionServiceClientBuilder
+        {
+            Endpoint = $"{locationId}-aiplatform.googleapis.com"
+        }.Build();
+
+        // Configure the parent resource.
+        var endpoint = EndpointName.FromProjectLocationPublisherModel(projectId, locationId, publisher, model);
+
+        var instance = new wkt::Value
+        {
+            StructValue = new()
+            {
+                Fields =
+                {
+                    ["messages"] = wkt::Value.ForList(
+                        wkt::Value.ForStruct(new()
+                        {
+                            Fields =
+                            {
+                                ["author"] = wkt::Value.ForString("user"),
+                                ["content"] = wkt::Value.ForString("Hi, how are you?"),
+                            }
+                        }),
+                        wkt::Value.ForStruct(new()
+                        {
+                            Fields =
+                            {
+                                ["author"] = wkt::Value.ForString("system"),
+                                ["content"] = wkt::Value.ForString("I am doing good. What can I help you in the coding world?"),
+                            }
+                        }),
+                        wkt::Value.ForStruct(new()
+                        {
+                            Fields =
+                            {
+                                ["author"] = wkt::Value.ForString("user"),
+                                ["content"] = wkt::Value.ForString("Please help write a function to calculate the min of two numbers."),
+                            }
+                        }))
+                }
+            }
+        };
+
+        var instances = new List<wkt::Value>
+        {
+            instance
+        };
+
+        var parameters = new wkt::Value
+        {
+            StructValue = new wkt::Struct
+            {
+                Fields =
+                {
+                    { "temperature", new wkt::Value { NumberValue = 0.3 } },
+                    { "maxOutputTokens", new wkt::Value { NumberValue = 1024 } }
+                }
+            }
+        };
+
+        // Make the request.
+        var response = client.Predict(endpoint, instances, parameters);
+
+        // Parse and return the content.
+        var content = response.Predictions[0].StructValue.Fields["candidates"].ListValue.Values[0].StructValue.Fields["content"].StringValue;
```

Comment:
This should be simplified after the library is generated from new protos. May be a TODO here as well. So, that we don't miss out on it later.

Reply:
I'll do another pass on all of these samples once the new library is out..so it's not really just here but the whole sample will be refactored.